### PR TITLE
`Exam mode`: Fix an issue with the result summary

### PR DESCRIPTION
**Pull request by edkaya**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).

### Motivation and Context
Fixes an issues introduced in #8974

### Description
* Fixes the params in a method call that is translated into a REST call to the server. Due to the wrong params, data could not be fetched and displayed in the user interface
* Additionally, removes some unused code

### Steps for Exam Mode Testing

Prerequisites:
- 1 Student
- 1 Instructor
- 1 Exam

1. Participate in the exam (student)
2. Publish the results (instructor)
3.  Check that the result summary is correctly displayed

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Exam Mode Test
- [x] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **New Features**
  - Introduced `studentExamId` in the `ExamResultSummaryComponent` to enhance grade information retrieval.

- **Refactor**
  - Simplified logic in `ExamParticipationComponent` by removing the `exerciseSelected` property.
  - Refined methods in `ExamParticipationService` for better dependency management and streamlined functionality.

### CODE REVIEW

File: src/main/webapp/app/exam/participate/summary/exam-result-summary.component.ts [hunk 1]
```diff
@@ -160,7 +164,7 @@ export class ExamResultSummaryComponent implements OnInit {
 
         if (isExamResultPublished(this.isTestRun, this.studentExam.exam, this.serverDateService)) {
             this.examParticipationService
-                .loadStudentExamGradeInfoForSummary(this.courseId, this.studentExam.exam.id, this.studentExam.user.id, this.studentExam.id)
+                .loadStudentExamGradeInfoForSummary(this.courseId, this.studentExam.exam.id, this.studentExam.id, this.studentExam.user.id)
```
Comment (krusche) [inline src/main/webapp/app/exam/participate/summary/exam-result-summary.component.ts:167]: this is the relevant code, everything else removes unused code

**Review by krusche — DISMISSED** [review]

**Review by krusche — APPROVED** [review]

**Review by florian-glombik — APPROVED** [review]
Code, double checked the unused code

**Review by florian-glombik — COMMENTED** [review]
Tested on staging, the summary is now displayed again

**Review by pzdr7 — APPROVED** [review]
Tested on TS3. Summary is displayed 👍

**Review by JohannesStoehr — APPROVED** [review]
Code

**Pull request merged**